Tests/IntegrationTests.Shared/AccessTests.cs

```diff
@@ -71,5 +95,19 @@ public void SetValueAndReplaceWithNull(string propertyName, object propertyValue
 
             Assert.That(TestHelpers.GetPropertyValue(ato, propertyName), Is.EqualTo(null));
         }
+
+        public IEnumerable<object[]> SetAndReplaceWithNullCases()
+        {
+            yield return new object[] { "NullableCharProperty", '0' };
```

Comment:
This test first sets the property to a non-null value and asserts the getter returns the value, and then sets the property to null and asserts that the getter returns null.